### DISCUSSION TITLE
Does A Volume's Replica(s) Influence Pod Scheduling?

### DISCUSSION THREAD

**Discussion by jsievenpiper** [longhorn/longhorn#7820] [Q&A]
I've recently started moving workloads to replicated longhorn volumes, and I'm trying to understand some behaviors that I couldn't quite grok from the documentation. I'm super happy to open a PR to update the documentation after sorting through this if it's helpful to others.

I have a standard deployment (single replica, if it matters) referencing a replicated longhorn volume. The volumes are spread across multiple nodes. If I restart the deployment, the pod will move around nodes appropriately -- but it doesn't _seem_ to be influenced by the existing longhorn volumes. Usually leading to longhorn creating _another_ volume replica, delaying pod spin up (and then subsequently destroying some other replica to bring the replica count back into check).

I have the volume locality set as `best-effort`, but these nodes have tons of capacity to them.

Is this expected? Everything is working otherwise, just seems odd to thrash volumes in that way -- admittedly I know very little about how the scheduler can be influenced!

Thanks in advance!

**Comment by c3y1huang** [comment]
Longhorn will try to maintain a replica on the same nodes as the workload when the volume's data locality is configured as `best-effort`. If you don't want this behavior, try setting it to `disabled`.

https://longhorn.io/docs/1.5.3/high-availability/data-locality/#data-locality-settings

**Reply by jsievenpiper** [reply]
Thanks! This is what I'm seeing, but what I'm really trying to confirm is if it has any effect the other way around: if I already have three replicas, when a pod is rescheduled, does longhorn influence which nodes are able to be chosen? e.g. I already have a volume replica on nodes A, B, and C -- often I find node D or E may be chosen (and a new replica rebuilt over there) rather than favoring an existing replica.
